### PR TITLE
improvements to add starter brick e2e spec

### PR DESCRIPTION
**Pull request by fungairino**
## What does this PR do?

- This PR refactors the starter brick end-to-end tests for the page editor to make it faster, more readable and more re-usable/consistent with POM usage.

### CODE REVIEW

File: end-to-end-tests/tests/pageEditor/starterBrickAdd.spec.ts
Comment (fungairino) [inline end-to-end-tests/tests/pageEditor/starterBrickAdd.spec.ts:1]: sidenote: should we rename this file to `addStarterBrick.spec.ts`

File: end-to-end-tests/pageObjects/pageEditor/pageEditorPage.ts [hunk 1]
```diff
@@ -100,25 +100,15 @@ export class PageEditorPage extends BasePageObject {
...
   /** Used for interactions that require selecting an element on the connected page, such as the button starter brick */
   @ModifiesModState
-  async selectConnectedPageElement(
-    connectedPage: Page,
-    selectLocator: Locator,
-    expectedElementSelector: string,
-  ) {
-    // Without focusing first, the click doesn't enable selection tool ¯\_(ツ)_/¯
-    await this.getByLabel("Select element").focus();
-    await this.getByLabel("Select element").click();
```
Comment (fungairino) [inline end-to-end-tests/pageObjects/pageEditor/pageEditorPage.ts:110]: This was only specific to the select element component in the brick config.

File: end-to-end-tests/tests/modLifecycle.spec.ts
Comment (fungairino) [inline end-to-end-tests/tests/modLifecycle.spec.ts:1]: made changes in this file to properly use the refactored `selectConnectedPageElement` element.

File: end-to-end-tests/tests/pageEditor/starterBrickAdd.spec.ts [hunk 2]
```diff
@@ -21,111 +21,95 @@ import { test, expect } from "../../fixtures/testBase";
 import { type Page, test as base } from "@playwright/test";
 import { getSidebarPage } from "../../utils";
 
-test("Add new Button starter brick", async ({ page, newPageEditorPage }) => {
```
Comment (fungairino) [inline end-to-end-tests/tests/pageEditor/starterBrickAdd.spec.ts:24]: Combined all the tests into one to improve total test runtime.

File: end-to-end-tests/tests/pageEditor/starterBrickAdd.spec.ts [hunk 3]
```diff
@@ -21,111 +21,95 @@ import { test, expect } from "../../fixtures/testBase";
...
+    await pageEditorPage.modListingPanel
+      .getByRole("button", {
+        name: "Start with a Template",
+      })
+      .click();
+
+    await expect(page).toHaveURL(templatesGalleryUrl);
+    const pageRequest = await pageRequestPromise;
+    const pageRequestResponse = await pageRequest.response();
+    expect(pageRequestResponse.status()).toBe(200);
```
Comment (fungairino) [inline end-to-end-tests/tests/pageEditor/starterBrickAdd.spec.ts:113]: This is related to the recent outage: We should verify that the page is actually loaded successfully.

File: end-to-end-tests/pageObjects/pageEditor/modListingPanel.ts [hunk 4]
```diff
@@ -66,6 +67,7 @@ export class ModListingPanel extends BasePageObject {
    * not the underlying type
    * @returns modName the generated mod name
    */
+  @ModifiesModState
```
Comment (mnholtz) [inline end-to-end-tests/pageObjects/pageEditor/modListingPanel.ts:70]: While we're at it, could you update the name of/comment for the `ModifiesModState` decorator please? I think this should be called something like `ModifiesModFormState` or something. The reason why I think it's relevant is that I forgot that this decorator really has to do with the form state/redux.

File: end-to-end-tests/pageObjects/pageEditor/pageEditorPage.ts [hunk 5]
```diff
@@ -100,25 +100,15 @@ export class PageEditorPage extends BasePageObject {
...
-
+  async selectConnectedPageElement(selectLocator: Locator) {
```
Comment (mnholtz) [inline end-to-end-tests/pageObjects/pageEditor/pageEditorPage.ts:103]:
```suggestion
  async selectInspectedPageElement(selectLocator: Locator) {
```

File: end-to-end-tests/tests/pageEditor/starterBrickAdd.spec.ts
Comment (mnholtz) [reply end-to-end-tests/tests/pageEditor/starterBrickAdd.spec.ts:1]: Sure

File: end-to-end-tests/tests/pageEditor/starterBrickAdd.spec.ts [hunk 6]
```diff
@@ -21,111 +21,95 @@ import { test, expect } from "../../fixtures/testBase";
...
-    await pageEditorPage.brickActionsPanel.getBricksInPipeline();
-  expect(brickPipeline).toHaveLength(2);
-  await expect(brickPipeline[0]).toContainText("Sidebar Panel");
-  await expect(brickPipeline[1]).toContainText("Render Document");
-  await expect(
-    pageEditorPage.brickConfigurationPanel.getByRole("textbox", {
-      name: "Name",
-    }),
-  ).toHaveValue("Sidebar Panel");
+    // TODO: save the mod and verify the snapshot of the mod definition with `verifyModDefinitionSnapshot` fixture
```
[the remark this answers is not in the document]
Comment (fungairino) [reply end-to-end-tests/tests/pageEditor/starterBrickAdd.spec.ts:46]: We would change this test once we remove standalone mods anyways, but we should be verifying the mod definition as part of these eventually.

File: end-to-end-tests/tests/pageEditor/starterBrickAdd.spec.ts [hunk 7]
```diff
@@ -21,111 +21,95 @@ import { test, expect } from "../../fixtures/testBase";
...
-  await expect(
-    pageEditorPage.brickConfigurationPanel.getByRole("textbox", {
-      name: "Name",
-    }),
-  ).toHaveValue("Quick Bar item");
-});
+    await expect(
+      pageEditorPage.getByText("My pbx.vercel.app button"),
+    ).toBeVisible();
+    await expect(brickPipeline).toHaveCount(1);
```
[the remark this answers is not in the document]
Comment (fungairino) [reply end-to-end-tests/tests/pageEditor/starterBrickAdd.spec.ts:38]: `toHaveCount` is only available for expect's with a locator argument that means how many elements were matched.

`toHaveLength` is a more generic, and is used with an array argument that was being returned from `.all()`

File: end-to-end-tests/pageObjects/pageEditor/pageEditorPage.ts (hunk 5 above)
Comment (fungairino) [reply end-to-end-tests/pageObjects/pageEditor/pageEditorPage.ts:103]: If we take this suggestion we should also rename a few other things in this class that reference the connected page to inspected page.

**Review by mnholtz — APPROVED** [review]

**Pull request merged**